sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -298,7 +304,10 @@ private void decodeDictionaryIds(
             // TODO: Convert dictionary of Binaries to dictionary of Longs
             if (!column.isNullAt(i)) {
               Binary v = dictionary.decodeToBinary(dictionaryIds.getDictId(i));
-              column.putLong(i, ParquetRowConverter.binaryToSQLTimestamp(v));
+              long rawTime = ParquetRowConverter.binaryToSQLTimestamp(v);
+              long adjTime =
```

Comment:
is it practical to consider caching the decoded and converted dictionary values?

Reply:
oh excellent point. we'd just need to store an additional `int -> long` map, but given that we've already got the dictionary this seems reasonable